core/ordering/cosipbft/mod_test.go

```diff
@@ -122,10 +114,74 @@ func TestService_Scenario_ViewChange(t *testing.T) {
 	err = nodes[1].pool.Add(makeTx(t, 0, nodes[1].signer))
 	require.NoError(t, err)
 
-	evt := waitEvent(t, events)
+	evt := waitEvent(t, events, DefaultTimeoutBeforeViewchange+2*time.Second)
 	require.Equal(t, uint64(0), evt.Index)
 }
 
+func TestService_Scenario_ViewChangeRequest(t *testing.T) {
+	nodes, ro, clean := makeAuthority(t, 4)
+	defer clean()
+	nodes[3].service.pool = fakePool{
+		Pool: nodes[3].service.pool,
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	err := nodes[1].service.Setup(ctx, ro)
+	require.NoError(t, err)
+
+	leader, err := nodes[0].service.pbftsm.GetLeader()
+	require.NoError(t, err)
+	require.Equal(t, leader, nodes[0].onet.GetAddress())
+
+	time.Sleep(DefaultRoundTimeout + 1*time.Second)
+
+	require.Equal(t, nodes[3].service.pbftsm.GetState(), pbft.ViewChangeState)
+	require.NotEqual(t, nodes[2].service.pbftsm.GetState(), pbft.ViewChangeState)
+	require.NotEqual(t, nodes[1].service.pbftsm.GetState(), pbft.ViewChangeState)
+	require.NotEqual(t, nodes[0].service.pbftsm.GetState(), pbft.ViewChangeState)
+
+	leader, err = nodes[0].service.pbftsm.GetLeader()
+	require.NoError(t, err)
+	// assert that node 0 is still the leader
+	require.Equal(t, leader, nodes[0].onet.GetAddress())
+}
+
+func TestService_Scenario_NoViewChangeRequest(t *testing.T) {
+	nodes, ro, clean := makeAuthority(t, 4)
+	defer clean()
+
+	signer := nodes[0].signer
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	initial := ro.Take(mino.RangeFilter(0, 4)).(crypto.CollectiveAuthority)
+
+	err := nodes[0].service.Setup(ctx, initial)
```

Comment:
why can't this just be 
```suggestion
	err := nodes[0].service.Setup(ctx, ro)
```
?